pvlib/pvsystem.py

```diff
@@ -511,6 +521,22 @@ def sapm_effective_irradiance(self, poa_direct, poa_diffuse,
             poa_direct, poa_diffuse, airmass_absolute, aoi,
             self.module_parameters, reference_irradiance=reference_irradiance)
 
+    def pvsyst_celltemp(self, poa_global, wind_speed, temp_air, eta_m=0.1,
+                        alpha_absorption=0.9):
```

Comment:
This does not follow the pattern for `PVSystem` methods. Instead of putting the kwargs in the method signature, the method should pull the values from the `self.module_parameters` dictionary. Please see other PVSystem methods for examples of using the `_build_kwargs` function.

Reply:
@wholmgren I looked at the other `PVSystem` implementations but am not sure how to work in your suggestion. The `self.module_parameters` attribute contains module parameters specific to SAPM with no analogues to `eta_m` or `alpha_absorption` that I'm aware of while the other parameters `poa_global`, `wind_speed` and `temp_air` are environmental.  

I modeled this off of `PVSystem.sapm_celltemp` which isn't in this format. Can you provide a little more guidance on how you see this working?

Reply:
@CameronTStark sorry I was unclear. 

`PVSystem.sapm_celltemp` is not a good template here because `pvsystem.sapm_celltemp`'s only keyword argument is explicitly set in `PVSystem` construction. Check out `PVSystem.physicaliam` for a simple example of how an attribute such as `PVSystem.module_parameters` may be referenced.

`poa_global`, `wind_speed` and `temp_air` should all be method arguments -- these are things that are external to the `PVSystem`.

`eta_m` and `alpha_absorption` describe the `PVSystem`, so they should be stored in an object attribute. In this case they should be stored in `module_parameters`. 

`module_parameters` is a general purpose attribute. It is not specific to SAPM or any other model.

`_build_kwargs` allows us to pull data specified in `module_parameters` if it's there. If it's not there, no kwarg is specified and we use the default values in the function definition.

Hope that helps, but let me know if not.

Reply:
Suggest using `PVSystem.ashraeiam` as the model. In the `pvsyst_celltemp` method, kwargs = `_build_kwargs(['eta_m', 'alpha_absorption'], self.module_parameters)` 
then
`return pvsyst_celltemp(poa_global, wind_speed, temp_air, temp_model=self.racking_model, **kwargs)`

Note the re-ordering of the arguments in the signature.

What will happen is that if a user has included `eta_m` and `alpha_absorption` in `module_parameters`, they will be extracted and passed in to `pvsystem.pvsyst_celltemp`.  If they aren't present, `pvsystem.pvsyst_celltemp` will use the default values.

Reply:
`PVSystem.sapm_celltemp` does not have this behavior currently for the `a` and `b` parameters it consumes.

Reply:
> PVSystem.sapm_celltemp does not have this behavior currently for the a and b parameters it consumes.

I don't follow. `a` and `b` are not explicit parameters in the function signature. `PVSystem.racking_model` is usually a string, but could be a list or dict as specified by the function `pvsystem.sapm_celltemp` function. I think some API changes could help, but that's a separate issue.

Reply:
I wasn't clear. "this behavior" I meant only that `PVSystem.sapm_celltemp` doesn't use `_build_kwargs` and isn't a model for what we're asking for in `PVSystem.pvsyst_celltemp`.  Agree that changing `sapm_celltemp` is separate from this PR.

Reply:
Great, thank you both for your guidance. I wasn't sure about the flexibility of `module_parameters` and now this all makes plenty of sense.